app/(main)/account/recovery/page.tsx

```diff
@@ -0,0 +1,263 @@
+    message: "Passwords don't match",
+    path: ['confirmPassword'],
+  });
+
+type ResetUserPasswordSchemaType = z.infer<typeof ResetPasswordSchema>;
+
+/**
+ * Renders the recover password page.
+ * @returns {React.JSX.Element} The rendered login page.
+ */
+const ResetPassword = (): React.JSX.Element => {
+  const router = useRouter();
+  const { isSignedIn, getUser } = useAuthContext();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  let userId: string | null = null;
+  let secret: string | null = null;
+  let expire: string | null = null;
+
+  /**
+   * Wrapper component to handle search params
+   * @returns {null} - null
+   */
+  const SearchParamsWrapper = (): null => {
+    const searchParams = useSearchParams();
+    userId = searchParams.get('userId');
+    secret = searchParams.get('secret');
+    expire = searchParams.get('expire');
+
+    useEffect(() => {
+      if (!userId || !secret || !expire) {
+        router.push('/login');
+        return;
+      }
+
+      setIsLoading(false);
+
+      if (isSignedIn) {
+        getUser();
+        router.push('/league/all');
+      }
+    }, [isSignedIn]);
+
+    return null;
+  };
+
+  const form = useForm<ResetUserPasswordSchemaType>({
+    resolver: zodResolver(ResetPasswordSchema),
+    defaultValues: {
+      password: '',
+      confirmPassword: '',
+    },
+  });
+
+  /**
+   * The current value of the 'password' field in the form.
+   * @type {string}
+   */
+  const password: string = useWatch({
+    control: form.control,
+    name: 'password',
+    defaultValue: '',
+  });
+
+  /**
+   * The current value of the 'confirmPassword' field in the form.
+   * @type {string}
+   */
+  const confirmPassword: string = useWatch({
+    control: form.control,
+    name: 'confirmPassword',
+    defaultValue: '',
+  });
+
+  /**
+   * Handles the form submission.
+   * @param data - The data from the form.
+   */
+  const onSubmit: SubmitHandler<ResetUserPasswordSchemaType> = async (data) => {
+    const { password } = data;
+    setIsSubmitting(true);
+
+    try {
+      await resetRecoveredPassword({
+        userId: userId as string,
+        token: secret as string,
+        password,
+      });
+
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Success}
+          message="Your password has been reset successfully! Please login with your new password."
+        />,
+      );
+
+      setTimeout(() => router.push('/login'), 2000);
+    } catch (err) {
+      console.error('Error resetting password:', err);
+      let errorMessage = 'There was an error resetting your password.';
+
+      if (err instanceof Error) {
+        const customErr = err as CustomError;
+        if (customErr.type === 'user_invalid_token') {
+          errorMessage =
+            'Invalid recovery token. Your token may have expired or already been used.  Please try again.';
```

Comment:
Why are we not using the error message that comes from the API instead of create our own?

Reply:
✅ updated to take the message from the API.